pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -131,8 +131,41 @@ type ReconcilePerconaServerMongoDB struct {
 	serverVersion *version.ServerVersion
 	reconcileIn   time.Duration
 
+	lockers lockStore
+}
+
+type lockStore struct {
+	mx    *sync.RWMutex
+	store map[string]lock
+}
+
+func newLockStore() lockStore {
+	return lockStore{
+		mx:    new(sync.RWMutex),
+		store: make(map[string]lock),
+	}
+}
+
+func (l lockStore) LoadOrCreate(key string) lock {
+	l.mx.RLock()
+	defer l.mx.RUnlock()
+
+	val, ok := l.store[key]
+	if !ok {
+		val = lock{
+			statusMutex: new(sync.Mutex),
+			updateSync:  new(int32),
+		}
+
+		l.store[key] = val
```

Comment:
You are doing `RLock` at the beginning, but here you are writing to map.
Worth to use lock for writing

Reply:
It is lock for writing

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/472/commits/24ffc577c757c987c74d55a94ff8032f218e7832
We do not need it at all, because we have sequentially run reconcile loops for PSMDB